internal/services/network/network_manager_verifier_workspace_resource.go

```diff
@@ -0,0 +1,264 @@
+	"github.com/hashicorp/go-azure-sdk/resource-manager/network/2024-05-01/ipampools"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/network/2024-05-01/verifierworkspaces"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var (
+	_ sdk.Resource           = ManagerVerifierWorkspaceResource{}
+	_ sdk.ResourceWithUpdate = ManagerVerifierWorkspaceResource{}
+)
+
+type ManagerVerifierWorkspaceResource struct{}
+
+func (ManagerVerifierWorkspaceResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return verifierworkspaces.ValidateVerifierWorkspaceID
+}
+
+func (ManagerVerifierWorkspaceResource) ResourceType() string {
+	return "azurerm_network_manager_verifier_workspace"
+}
+
+func (ManagerVerifierWorkspaceResource) ModelObject() interface{} {
+	return &ManagerVerifierWorkspaceResourceModel{}
+}
+
+type ManagerVerifierWorkspaceResourceModel struct {
+	Description      string                 `tfschema:"description"`
+	Location         string                 `tfschema:"location"`
+	Name             string                 `tfschema:"name"`
+	NetworkManagerId string                 `tfschema:"network_manager_id"`
+	Tags             map[string]interface{} `tfschema:"tags"`
+}
+
+func (ManagerVerifierWorkspaceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9\_\.\-]{1,64}$`),
+				"name must be between 1 and 64 characters long and can only contain letters, numbers, underscores(_), periods(.), and hyphens(-).",
+			),
+		},
+
+		"network_manager_id": commonschema.ResourceIDReferenceRequiredForceNew(&ipampools.NetworkManagerId{}),
+
+		"location": commonschema.Location(),
+
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (ManagerVerifierWorkspaceResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ManagerVerifierWorkspaceResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Network.VerifierWorkspaces
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var config ManagerVerifierWorkspaceResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			networkManagerId, err := verifierworkspaces.ParseNetworkManagerID(config.NetworkManagerId)
+			if err != nil {
+				return fmt.Errorf("parsing `network_manager_id`: %+v", err)
```

Comment:
These shouldn't be wrapped since the error messages already contain a lot of information so this information is redundant
```suggestion
				return err
```

Reply:
updated